enclave_wrangler/dataset_upload.py

```diff
@@ -34,11 +35,131 @@
+                #  ...Leaving this blank for now since is optional param.
+                # current_max_version=version_d[''],
+                # todo: concept_set: is this correct?
+                concept_set=version_d['concept_set_name'],  # == container_d['concept_set_name']
+                # todo: annotation: this should be moved into the new palantir-3-file data model, whatever that is
+                annotation='Curated value set: ' + version_d['concept_set_name'],
+                limitations=version_d['limitations'],
+                intention=version_d['intention'],
+                # todo: base_version: @Siggie: Leaving out, since is optional param, and not possible to include without
+                #  first adding current_max_version, for which more time is needed to figure out how to reliably ascertain.
+                # base_version=1,
+                # todo: intended_research_project: (a) default this to ENCLAVE_PROJECT_NAME in func, (b) do that here, (c) add it as a column to an
+                #  updated palantir-3-file for the new api
+                intended_research_project=ENCLAVE_PROJECT_NAME,
+                version_id=version_d['version'],
+                # todo: authority: @Siggie: Not sure what to put here, but it is optional param, so I'm leaving blank
+                #  - we didn't have this in the old palantir-3-file data model, but should add to the new one
+                # authority=version_d[''],
+                # TODO: validate first (if this func says validate_first=True, then upload
+                validate=True)
+
+            for item_d in version['items']:
+                # TODO: validate first (if this func says validate_first=True, then upload
+                # TODO: Not finished. How to actually add the concepts to existing concept set version?
+                #  ...look at comments above new_enclave_api.upload_concept() for some alternative endpoints, etc
+                # TODO: When doing w/ 'validate=False', I got the following error:
+                #  Unexpected <class 'requests.exceptions.HTTPError'>: 400 Client Error: Bad Request for url: https://unite.nih.gov/api/v1/ontologies/ri.ontology.main.ontology.00000000-0000-0000-0000-000000000000/actions/set-omop-concept-set-version-item/apply
+                response = upload_concept(  # concept_set_version_items / expressions
+                    include_descendants=item_d['includeDescendants'],
+                    # todo: concept_set_version_item: What to put here? (item_id is a random uuid)
+                    concept_set_version_item=item_d['item_id'],
+                    is_excluded=item_d['isExcluded'],
+                    include_mapped=item_d['includeMapped'],
+                    # TODO: validate first (if this func says validate_first=True, then upload
+                    validate=True)
+
+    # TODO: Siggie: updating these files is probably not worth / necessary right now
```

Comment:
@Sigfried Left a note in the code: `# TODO: Siggie: updating these files is probably not worth / necessary right now`

I agree with that in theory, but let's see how it works in practice. I think we no longer need to (i) update our `termhub-csets/cset_registry.csv`, as well as (ii) the need to update our local files that we're uploading with any values that we did not assign ourselves, but were auto-assigned by the enclave. 

That is, we no longer need to do this, so long as we download complete updates from the enclave. My only concern is that those files are very large, and it could take some time are able to implement a way to do frequent fetches.